javascript/ql/src/semmle/javascript/frameworks/Babel.qll

```diff
@@ -152,4 +164,33 @@ module Babel {
       result = pathExpr.getConfig().getFolder()
     }
   }
+
+  /**
+   * A configuration object for the `transform-react-jsx` plugin.
+   *
+   * The plugin option `{"pragma": xxx}` specifies a variable name used to instantiate
+   * JSX elements.
+   */
+  class TransformReactJsxConfig extends JSONArray {
+    Config cfg;
+
+    TransformReactJsxConfig() {
+      this = cfg.getPluginConfig("transform-react-jsx")
+    }
+
+    /** Gets the Babel configuration object. */
+    Config getConfig() {
+      result = cfg
+    }
+
+    /** Gets the options passed to this plugin. */
+    JSONObject getOptions() {
+      result = getElementValue(1)
+    }
+
+    /** Gets the name of the variable used to create JSX elements. */
+    string getJSXFactoryVariableName() {
```

Comment:
Perhaps change `JSX` to `Jsx` for consistency with the class name (and our normal camel-case conventions).